### PR TITLE
[DOCS] Add 'Using ES|QL in Elastic Security'

### PR DESCRIPTION
**Pull request by abdonpijpelink**
Adds a 'Using ES|QL in Elastic Security' section, similar to 'Using ES|QL in Kibana'.

### CODE REVIEW

**Comment by abdonpijpelink** [comment]
@elasticmachine test this please

**Review by nastasha-solomon — APPROVED** [review]
Thanks for putting this together, @abdonpijpelink! The links work for now, but I might want to change them after @benironside and I merge our ES|QL PRs. That way, they point to ES|QL docs instead of the top-level Timeline, rule, and AI Assistant docs.

**Comment by abdonpijpelink** [comment]
> The links work for now, but I might want to change them after @benironside and I merge our ES|QL PRs. That way, they point to ES|QL docs instead of the top-level Timeline, rule, and AI Assistant docs.

I'll wait for your PRs to merge, after which I'll update and merge this PR.

File: docs/reference/esql/esql-security-solution.asciidoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+more, refer to {security-guide}/timelines-ui.html[Investigate events in
+Timeline].
+
+[discrete]
+[[esql-elastic-security-detection-rules]]
+=== Use {esql} to create detection rules
+
+Use the {esql} rule type to create detection rules using {esql} queries. The
+{esql} rule type supports aggregating and non-aggregating queries. To learn
+more, refer to {security-guide}/rules-ui-create.html[Create a detection rule].
```
Comment (nastasha-solomon) [inline docs/reference/esql/esql-security-solution.asciidoc:27]:
```suggestion
more, refer to {security-guide}/rules-ui-create.html#create-esql-rule[Create an {esql} rule].
```

**Pull request merged**